linker/shared/src/main/scala/org/scalajs/linker/checker/IRChecker.scala

```diff
@@ -737,11 +737,35 @@ private final class IRChecker(unit: LinkingUnit, reporter: ErrorReporter,
           override def traverse(tree: Tree): Unit = typecheck(tree, env)
         })
 
-      case _: RecordSelect if postOptimizer =>
-        // TODO
+      case RecordSelect(record, SimpleFieldIdent(fieldName)) if postOptimizer =>
+        record.tpe match {
+          case NothingType => // ok
+
+          case RecordType(fields) =>
+            fields.find(_.name == fieldName) match {
+              case Some(field) =>
+                if (tree.tpe != field.tpe)
+                  reportError(i"Record select of field type ${field.tpe} typed as ${tree.tpe}")
+
+              case None =>
+                reportError(i"Record select of non-existent field: $fieldName")
+            }
+
+          case tpe =>
+            reportError(i"Record type expected but $tpe found")
+        }
+
+        typecheck(record, env)
 
-      case _: RecordValue if postOptimizer =>
-        // TODO
+      case RecordValue(RecordType(fields), elems) if postOptimizer =>
+        if (fields.size == elems.size) {
+          for ((field, elem) <- fields.zip(elems))
+            typecheckExpect(elem, env, field.tpe)
+        } else {
+          reportError(
+              "Mismatched size for record fields / elements: " +
+              i"${fields.size} fields vs ${elems.size} elements")
+        }
```

Comment:
I think in the `else` branch we should have something like
```scala
for (elem <- elems)
  typecheckExpr(elem)
```
Otherwise we'll miss follow-up errors inside the `elems`.

It's not critical because IR checking will fail anyway, but I think we normally do this in other similar situations. For example in `Apply`.